### PR TITLE
basic powershell support

### PR DESCRIPTION
**Pull request by aaroneg**
I got the build issues sorted out, but probably not in the best way possible. At any rate, the lexer works pretty well.

### CODE REVIEW

**Comment by jneen** [comment]
Hi!  Sorry I've been out of commission for a little while, and I'm just getting some time to work on rouge. Thanks so much for the lexer, and for taking the time to get the build green.

The one issue I see is that it looks largely copy-pasted from the `shell` lexer - would you be willing to refactor it so it's a subclass of the `shell` lexer using the `prepend`/`append` interface? (see C vs C++/ObjC or QML vs Javascript f. ex).

**Comment by aaroneg** [comment]
It is indeed a copy / modification of shell. I'm not especially familiar with Ruby or Rouge in general and I couldn't find much in the way of documentation aside from existing lexers + trial & error. My goal is to get powershell support in gitlab, by way of rouge->rugments. I'll have a look at the stuff you mentioned and see what I can do.

**Comment by jneen** [comment]
Cool! Yeah it's pretty straightforward once you get into it. Let me know if you have any questions.

**Comment by aaroneg** [comment]
@jneen - I think i've got it done. Have a look : https://github.com/aaroneg/rugments/blob/master/lib/rugments/lexers/powershell.rb

**Comment by aaroneg** [comment]
@jneen - https://github.com/aaroneg/rouge/blob/master/lib/rouge/lexers/powershell.rb 
Does that match what you were asking for?

**Comment by jneen** [comment]
Awesome! Just a few style fixes and I'll merge it. (comments forthcoming)

File: lib/rouge/lexers/powershell.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    load_const :Shell, 'shell.rb'
+  
+      class Powershell < Shell
```
Comment (jneen) [inline lib/rouge/lexers/powershell.rb]: dedent

File: lib/rouge/lexers/powershell.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      Begin Exit Process Break Filter Return Catch Finally Sequence Class For Switch Continue ForEach Throw Data From Trap Define Function Try Do If Until DynamicParam In Using Else InlineScript Var ElseIf Parallel While End Param Workflow 
+      ).join('|')
+    
+      KEYWORDS_TYPE = %w(
+        bool byte char decimal double float int long object sbyte
+        short string uint ulong ushort
+      ).join('|')
+    
+      OPERATORS = %w(
+        -split -isplit -csplit -join -is -isnot -as -eq -ieq -ceq -ne -ine -cne -gt -igt -cgt -ge -ige -cge -lt -ilt -clt -le -ile -cle -like -ilike -clike -notlike -inotlike -cnotlike -match -imatch -cmatch -notmatch -inotmatch -cnotmatch -contains -icontains -ccontains -notcontains -inotcontains -cnotcontains -replace -ireplace -creplace -band -bor -bxor -and -or -xor \. & = \+= -= \*= \/= %=
```
Comment (jneen) [inline lib/rouge/lexers/powershell.rb]: break into 80-char lines

File: lib/rouge/lexers/powershell.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
...roperty Resolve-Path Restart-Service Resume-Service Select-Object Select-String Set-Acl Set-Alias Set-AuthenticodeSignature Set-Content Set-Date Set-ExecutionPolicy Set-Item Set-ItemProperty Set-Location Set-PSDebug Set-Service Set-TraceSource Set-Variable Sort-Object Split-Path Start-Service Start-Sleep Start-Transcript Stop-Process Stop-Service Stop-Transcript Suspend-Service Tee-Object Test-Path Trace-Command Update-FormatData Update-TypeData Where-Object Write-Debug Write-Error Write-Host Write-Output Write-Progress Write-Verbose Write-Warning ac asnp cat cd chdir clc clear clhy cli clp cls clv cnsn compare copy cp cpi cpp curl cvpa dbp del diff dir dnsn ebp echo epal epcsv epsn erase etsn exsn fc fl foreach ft fw gal gbp gc gci gcm gcs gdr ghy gi gjb gl gm gmo gp gps group gsn gsnp gsv gu gv gwmi h history icm iex ihy ii ipal ipcsv ipmo ipsn irm ise iwmi iwr kill lp ls man md measure mi mount move mp mv nal ndr ni nmo npssc nsn nv ogv oh popd ps pushd pwd r rbp rcjb rcsn rd rdr ren ri rjb rm rmdir rmo rni rnp rp rsn rsnp rujb rv rvpa rwmi sajb sal saps sasv sbp sc select set shcm si sl sleep sls sort sp spjb spps spsv start sujb sv swmi tee trcm type wget where wjb write \% \?
```
Comment (jneen) [inline lib/rouge/lexers/powershell.rb]: indent, 80-char lines

File: lib/rouge/demos/powershell [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# If not running interactively, don't do anything
+[[ -z "$PS1" ]] && return
```
Comment (jneen) [inline lib/rouge/demos/powershell]: could you add a demo that shows off some of the powershell-specific features?

**Comment by aaroneg** [comment]
How does that look?

**Comment by jneen** [comment]
you know what i'll just fix it.

**Comment by jneen** [comment]
Thanks for the lexer!

**Comment by aaroneg** [comment]
thanks! glad you like.

**Comment by jneen** [comment]
this is more or less what i meant: https://github.com/jneen/rouge/commit/57cfd27865fc8163699348e9538a4ce8214ce9f8

**Comment by aaroneg** [comment]
Thanks, I don't speak ruby so I appreciate the help.

**Pull request merged**